src/TfvcMigrator/Program.cs

```diff
@@ -68,273 +68,285 @@ private static RootPathChange ParseRootPathChange(string token)
         ImmutableArray<RootPathChange> rootPathChanges = default,
         string? pat = null)
     {
-        if (rootPathChanges.IsDefault) rootPathChanges = ImmutableArray<RootPathChange>.Empty;
+        try
+        {
+            if (rootPathChanges.IsDefault) rootPathChanges = ImmutableArray<RootPathChange>.Empty;
 
-        var outputDirectory = Path.GetFullPath(
-            new[] { outDir, PathUtils.GetLeaf(rootPath), projectCollectionUrl.Segments.LastOrDefault() }
-                .First(name => !string.IsNullOrEmpty(name))!);
+            var outputDirectory = Path.GetFullPath(
+                new[] { outDir, PathUtils.GetLeaf(rootPath), projectCollectionUrl.Segments.LastOrDefault() }
+                    .First(name => !string.IsNullOrEmpty(name))!);
 
-        using var repo = InitRepository(outputDirectory);
-        if (repo is null)
-            return 1;
+            using var repo = InitRepository(outputDirectory);
+            if (repo is null)
+                return 1;
 
-        var authorsLookup = LoadAuthors(authors);
+            await using var authorsFileStream = LoadAuthorsFileStream(authors);
```

Comment:
I'm not sure about holding the file open until the migration ends. Could we extract all the stuff that needs the file to a new method so that we're sure the file isn't open when that method returns?